gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -268,5 +285,17 @@ public void clearActiveProject() {
     cache.clearActiveProject();
     localValueStore.clearLastActiveProjectId();
     activeProject.onNext(Persistable.notLoaded());
+    localDataStore
+        .getLastActiveProject()
+        .doOnSuccess(project -> localDataStore.removeProject(project).subscribe())
```

Comment:
Instead of using doOnSuccesss here, I believe you should be able to use flat map keeping one level of streams instead of nesting (since the Maybe emits a project):

```java
.getLastActiveProject()
.flatMapCompleteable(localDataStore::removeProject)
.subscribe();
```

Or roughly that http://reactivex.io/RxJava/javadoc/io/reactivex/Maybe.html#flatMapCompletable-io.reactivex.functions.Function-

Also, just subscribing here seems fishy to me, but it's been a while since I've used Rx so it may be totally fine.

Reply:
+1.  The call the subscribe() would generate a new subscription which would need to be kept and properly disposed of. By chaining streams this is handled for you.

However, since this is an async operation, you'd then need to return the stream to the caller so that it can subscribe() and dispose properly as part of the project activation stream.

Reply:
Thanks!